osu.Game/Models/RealmBeatmapSet.cs

```diff
@@ -53,25 +53,30 @@ public class RealmBeatmapSet : RealmObject, IHasGuidPrimaryKey, IHasRealmFiles,
         /// <param name="filename">The name of the file to get the storage path of.</param>
         public string? GetPathForFile(string filename) => Files.SingleOrDefault(f => string.Equals(f.Filename, filename, StringComparison.OrdinalIgnoreCase))?.File.StoragePath;
 
-        public override string ToString() => Metadata?.ToString() ?? base.ToString();
-
         public bool Equals(RealmBeatmapSet? other)
         {
-            if (other == null)
-                return false;
+            if (ReferenceEquals(this, other)) return true;
+            if (other == null) return false;
 
-            if (IsManaged && other.IsManaged)
-                return ID == other.ID;
+            return ID == other.ID;
+        }
+
+        public bool Equals(IBeatmapSetInfo? other)
+        {
+            if (ReferenceEquals(this, other)) return true;
+            if (other == null) return false;
+
+            if (other is RealmBeatmapSet b && Equals(b))
+                return true;
 
             if (OnlineID > 0 && other.OnlineID > 0)
                 return OnlineID == other.OnlineID;
 
-            if (!string.IsNullOrEmpty(Hash) && !string.IsNullOrEmpty(other.Hash))
-                return Hash == other.Hash;
-
-            return ReferenceEquals(this, other);
+            return false;
         }
 
+        public override string ToString() => Metadata?.ToString() ?? base.ToString();
```

Comment:
Bit of a random addition but I'd probably prefer the new extension `GetDisplayString()`.